### PR TITLE
 Handle Serv-U answer when trying to delete a directory

### LINKED ISSUE #35: Can't remove directories on a Serv-U server
Commit e8b14451d6c62d63aa3d5dab67fcbd11e7740381 merged RemoveDirectory() into the general Remove() call and tries to be smart by attempting an RMDIR call if REMOVE fails with an a FAILURE code.

This works ok for compliant servers, but Serv-U returns 0x18 when issuing a REMOVE call on a directory and Remove() fails as a result.

I'm not sure it really makes sense to merge RemoveFile and RemoveDirectory into one call anyway; it doesn't match the underlying protocol - Could we consider reversing that?  Or at least adding an explicit RemoveDirectory() call if backwards compatibility is a concern.

### PR DESCRIPTION
**Pull request by xiu**
Fixes #35 

Added status code from http://tools.ietf.org/html/draft-ietf-secsh-filexfer-13 and a new condition to Client.Remove()

### CODE REVIEW

File: sftp.go [hunk 1]
```diff
@@ -37,15 +37,38 @@ const (
...
+	ssh_FX_WRITE_PROTECT               = 12
+	ssh_FX_NO_MEDIA                    = 13
+	ssh_FX_NO_SPACE_ON_FILESYSTEM      = 14
+	ssh_FX_QUOTA_EXCEEDED              = 15
+	ssh_FX_UNKNOWN_PRINCIPAL           = 16
+	ssh_FX_LOCK_CONFLICT               = 17
+	ssh_FX_DIR_NOT_EMPTY               = 18
+	ssh_FX_NOT_A_DIRECTORY             = 19
+	ssh_FX_INVALID_FILENAME            = 20
+	ssh_FX_LINK_LOOP                   = 21
```
Comment (davecheney) [inline sftp.go]: If any of these come from an rfc or draft RFCs could you please add a line or block comment

// see rfc - draft nnnn $url

Or something

**Comment by davecheney** [comment]
LGTM, can you please address the one nit and update the PR description to

Fixes #35 

Don't worry about the travis Ci failure, I haven't got that build passing yet.

**Comment by xiu** [comment]
Sure, done :)

**Comment by davecheney** [comment]
Lovely, thanks

**Comment by davecheney** [comment]
Fixes #35

**Pull request merged**